client/rpc-servers/src/middleware.rs

```diff
@@ -48,7 +48,7 @@ impl RpcMetrics {
 					&["protocol"]
 				).ok()?, r).ok()?,
 			})
-		}).ok_or(PrometheusError::Msg("Cannot register metric".to_string()))
+		}).ok_or(PrometheusError::AlreadyReg)
```

Comment:
While this is not your fault. This function is not making any sense. Basically it requires that we pass a registry object :facepalm:

Reply:
I think that's a topic for another PR

Reply:
Yeah